src/utils/utils.tsx

```diff
@@ -81,3 +85,77 @@ export const getNetworkPath = (network?: Network): string => {
     return `https://${network}.tradetrust.io`;
   }
 };
+
+export const getIssuerLocation = (rawDocument: OpenAttestationDocument): string | undefined => {
+  if (utils.isRawV2Document(rawDocument)) {
+    const { issuers } = rawDocument;
+    return issuers[0].identityProof?.location; // let's assume only 1 issuer, so far no multiple issuer cases
+  } else if (utils.isRawV3Document(rawDocument)) {
+    return rawDocument.openAttestationMetadata.identityProof.identifier;
+  }
+  throw new Error(
+    "Unsupported document type: Only can retrieve issuer location from OpenAttestation v2 & v3 documents."
+  );
+};
+
+export const getIdentityProofType = (rawDocument: OpenAttestationDocument): string | undefined => {
+  if (utils.isRawV2Document(rawDocument)) {
+    const { issuers } = rawDocument;
+    return issuers[0].identityProof?.type; // let's assume only 1 issuer, so far no multiple issuer cases
+  } else if (utils.isRawV3Document(rawDocument)) {
+    return rawDocument.openAttestationMetadata.identityProof.type;
+  }
+  throw new Error(
+    "Unsupported document type: Only can retrieve IdentityProof type from OpenAttestation v2 & v3 documents."
+  );
+};
+
+export const getIssuerAddress = (rawDocument: OpenAttestationDocument): string | undefined => {
+  if (utils.isRawV2Document(rawDocument)) {
+    const { issuers } = rawDocument;
+    return (
+      issuers[0].certificateStore ||
+      issuers[0].documentStore ||
+      issuers[0].tokenRegistry ||
+      issuers[0].identityProof?.key
+    ); // let's assume only 1 issuer, so far no multiple issuer cases
+  } else if (utils.isRawV3Document(rawDocument)) {
+    return rawDocument.openAttestationMetadata.proof.value;
+  }
+  throw new Error(
+    "Unsupported document type: Only can retrieve issuer address from OpenAttestation v2 & v3 documents."
+  );
+};
```

Comment:
i see that the code is a little from the original code. what if its DNSDID document? cos if i rmb correctly, later on in the issuing part it will check for the address or the string "DNS-DID" to do its issuing accordingly.

Reply:
didn't quite understand

Reply:
ok got it after discussion